spyder_line_profiler/spyder/widgets.py

```diff
@@ -100,123 +100,214 @@ def natural_sort_key(key):
+        self.log_action = self.create_action(
+            SpyderLineProfilerWidgetActions.ShowOutput,
+            text=_("Show Result"),
             tip=_("Show program's output"),
-            triggered=self.show_log)
-
-        self.datatree = LineProfilerDataTree(self)
-
-        self.collapse_button = create_toolbutton(
-            self,
-            icon=get_icon('collapse.png'),
+            icon=self.create_icon('log'),
+            triggered=self.show_log,
+        )
+        self.collapse_action = self.create_action(
+            SpyderLineProfilerWidgetActions.Collapse,
+            text=_("Collaps"),
+            tip=_('Collapse all'),
+            icon=self.create_icon('collapse'),
             triggered=lambda dD=-1: self.datatree.collapseAll(),
-            tip=_('Collapse all'))
-        self.expand_button = create_toolbutton(
-            self,
-            icon=get_icon('expand.png'),
-            triggered=lambda dD=1: self.datatree.expandAll(),
-            tip=_('Expand all'))
-
-        hlayout1 = QHBoxLayout()
-        hlayout1.addWidget(self.filecombo)
-        hlayout1.addWidget(browse_button)
-        hlayout1.addWidget(self.start_button)
-        hlayout1.addWidget(self.stop_button)
-
-        hlayout2 = QHBoxLayout()
-        hlayout2.addWidget(self.collapse_button)
-        hlayout2.addWidget(self.expand_button)
-        hlayout2.addStretch()
-        hlayout2.addWidget(self.datelabel)
-        hlayout2.addStretch()
-        hlayout2.addWidget(self.log_button)
+        )
+        self.expand_action = self.create_action(
+            SpyderLineProfilerWidgetActions.Expand,
+            text=_("Expand"),
+            tip=_('Expand all'),
+            icon=self.create_icon('expand'),
+            triggered=lambda dD=-1: self.datatree.expandAll(),
+        )
+        self.save_action = self.create_action(
+            SpyderLineProfilerWidgetActions.SaveData,
+            text=_("Save data"),
+            tip=_('Save line profiling data'),
+            icon=self.create_icon('filesave'),
+            triggered=self.save_data,
+        )
+        self.clear_action = self.create_action(
+            SpyderLineProfilerWidgetActions.Clear,
+            text=_("Clear output"),
+            tip=_('Clear'),
+            icon=self.create_icon('editdelete'),
+            triggered=self.clear_data,
+        )
 
-        layout = QVBoxLayout()
-        layout.addLayout(hlayout1)
-        layout.addLayout(hlayout2)
-        layout.addWidget(self.datatree)
-        self.setLayout(layout)
-
-        self.process = None
         self.set_running_state(False)
-        self.start_button.setEnabled(False)
-
+        self.start_action.setEnabled(False)
+        self.clear_action.setEnabled(False)
+        self.log_action.setEnabled(False)
+        self.save_action.setEnabled(False)
+        
+        # Main Toolbar
+        toolbar = self.get_main_toolbar()
+        for item in [self.filecombo, self.browse_action, self.start_action, 
```

Comment:
Remove trailing whitespace:
```suggestion
        for item in [self.filecombo, self.browse_action, self.start_action,
```